### PR TITLE
[browser] introduce sourcemaps

### PR DESCRIPTION
**Pull request by pavelsavara**
For `dotnet.runtime.js` and `dotnet.js`, which are built from our typescript.
- It will link `file:///C:/Dev/runtime/src/mono/wasm/runtime/loader/logging.ts` while running on local machine. 
    - This doesn't work on FF, but it does on Chrome.
- It will [link](https://raw.githubusercontent.com/dotnet/runtime/a146e4e3a58e7287fe831164b3ee273b83c52758/src/mono/wasm/runtime/loader/logging.ts) `https://raw.githubusercontent.com/dotnet/runtime/a146e4e3a58e7287fe831164b3ee273b83c52758/src/mono/wasm/runtime/loader/logging.ts` when built on CI, with the correct git hash.
- public MSBuild property `WasmEmitSourceMap`, defaults to true for build and false for publish

### CODE REVIEW

File: src/mono/wasm/runtime/rollup.config.js [hunk 1]
```diff
@@ -72,6 +76,25 @@ try {
     gitHash = "unknown";
 }
 
+const locationCache = {};
+function sourcemapPathTransform(relativeSourcePath, sourcemapPath) {
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/rollup.config.js:80]: How/where is this function called?

**Review by ilonatommy — APPROVED** [review]

**Review by maraf — CHANGES_REQUESTED** [review]
It should have build time configuration for propagation to the build output folder (I believe off by default, or at least for publish)

File: src/mono/wasm/runtime/rollup.config.js (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/rollup.config.js:80]: Line 133, it's a rollup plugin

**Comment by pavelsavara** [comment]
> It should have build time configuration for propagation to the build output folder (I believe off by default, or at least for publish)

Done via `WasmEmitSourceMap`

**Review by maraf — APPROVED** [review]
Looks good to me 👍

**Pull request merged**